Libplanet/Assets/Currency.cs

```diff
@@ -61,19 +62,67 @@ namespace Libplanet.Assets
         /// </summary>
         public readonly HashDigest<SHA1> Hash;
 
+        /// <summary>
+        /// Whether the total supply of this instance of <see cref="Currency"/> is trackable.
+        /// </summary>
+        public readonly bool TotalSupplyTrackable;
+
+        private readonly (BigInteger Major, BigInteger Minor)? _maximumSupply;
+
         /// <summary>
         /// Defines a <see cref="Currency"/> type.
         /// </summary>
         /// <param name="ticker">The ticker symbol, e.g., <c>&quot;USD&quot;</c>.</param>
         /// <param name="decimalPlaces">The number of digits to treat as <a
         /// href="https://w.wiki/ZXv#Treatment_of_minor_currency_units_(the_%22exponent%22)">minor
         /// units (i.e., exponent)</a>.</param>
+        /// <param name="maximumSupply">The uppermost quantity of currency allowed to exist.
+        /// See also <see cref="MaximumSupply"/> field which corresponds to this.
+        /// Must be <value>null</value> when <paramref name="totalSupplyTrackable" /> is
+        /// <value>null</value>.</param>
         /// <param name="minters">The <see cref="Address"/>es who can mint the currency.
         /// See also <see cref="Minters"/> field which corresponds to this.</param>
+        /// <param name="totalSupplyTrackable">Whether the total supply of this currency is
+        /// trackable. This parameter is for backwards compatibility with
+        /// <see cref="Currency">Currencies</see> defined before maximum supply was introduced,
+        /// and must be set to <value>false</value> in that case, and should be <value>true</value>
+        /// for new <see cref="Currency">Currencies</see>.</param>
```

Comment:
As its parameter type is unusual, users could be confused what's major and what's minor and what values should be passed to them.  How about providing an example?  Like this:

> For example, `maximumSupply` in `new Currency("ABC", 2, (123, 45))` means token ABC can be minted up to 123.45 ABC.

Reply:
That would make more sense. 👍